samples/templates/default-azuredeploy.json

```
@@ -193,9 +193,10 @@
             "DicomServer__Features__EnableExtendedQueryTags": "true",
             "DicomServer__Security__Enabled": "[variables('securityAuthenticationEnabled')]",
             "DicomServer__Security__Authentication__Authority": "[parameters('securityAuthenticationAuthority')]",
-            "DicomServer__Security__Authentication__Audience": "[parameters('securityAuthenticationAudience')]"
+            "DicomServer__Security__Authentication__Audience": "[parameters('securityAuthenticationAudience')]",
+            "KeyVault__Endpoint": "[variables('keyVaultEndpoint')]"
```

I think we can directly use the `reference` function with the [`vaultUri`](https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/vaults?tabs=bicep) property. Although we may need to wait until it has been created?

```suggestion
            "KeyVault__Endpoint": "[reference(resourceId('Microsoft.KeyVault/vaults', variables('serviceName'))).vaultUri]"
```

Or later we have to dynamically add the JSON like: `[reference(variables('keyVaultResourceId')).vaultUri]`